include/ddc/kernels/splines/spline_builder.hpp

```diff
@@ -329,7 +329,11 @@ class SplineBuilder
      */
     batched_spline_tr_domain_type batched_spline_tr_domain() const noexcept
     {
-        return batched_spline_tr_domain_type(spline_domain(), batch_domain());
+        return batched_spline_tr_domain_type(
+                batched_spline_domain().restrict(ddc::DiscreteDomain<bsplines_type>(
+                        ddc::DiscreteElement<bsplines_type>(0),
+                        ddc::DiscreteVector<bsplines_type>(
+                                ddc::discrete_space<bsplines_type>().nbasis()))));
```

Comment:
Just to understand why it works with or without this change. This affects spline dimension which is the non-contiguous dimension in the linear system right ?

Reply:
I completely agree but I think it is not only a question about extents, looking at the source after this transpose loop we have
```cpp
    Kokkos::View<double**, Kokkos::LayoutRight, exec_space> bcoef_section(
            spline_tr.data_handle(),
            ddc::discrete_space<bsplines_type>().nbasis(),
            batch_domain().size());
```

If the dimension of interest would have been contiguous, this `bcoef_section` would not have been a valid view of the `spline_tr` array.